src/spawn-win.c

```diff
@@ -1,10 +1,75 @@
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <windows.h> 
+#include <strsafe.h>
+
 #include "include/mini/common.h"
 
+static char executable_path[BUFSIZE];
+
+int popen_complete(const char* cmd)
+{
+  STARTUPINFO StartupInfo;
+  ZeroMemory(&StartupInfo, sizeof(StartupInfo));
+  StartupInfo.cb = sizeof(StartupInfo);
+
+  PROCESS_INFORMATION ProcessInformation;
+  ZeroMemory(&ProcessInformation, sizeof(ProcessInformation));
+
+  //  BOOL WINAPI CreateProcess(
+  //    _In_opt_    LPCTSTR               lpApplicationName,
+  //    _Inout_opt_ LPTSTR                lpCommandLine,
+  //    _In_opt_    LPSECURITY_ATTRIBUTES lpProcessAttributes,
+  //    _In_opt_    LPSECURITY_ATTRIBUTES lpThreadAttributes,
+  //    _In_        BOOL                  bInheritHandles,
+  //    _In_        DWORD                 dwCreationFlags,
+  //    _In_opt_    LPVOID                lpEnvironment,
+  //    _In_opt_    LPCTSTR               lpCurrentDirectory,
+  //    _In_        LPSTARTUPINFO         lpStartupInfo,
+  //    Out_       LPPROCESS_INFORMATION lpProcessInformation
+  //  );
+  BOOL ret = CreateProcess(
+    NULL,                     // No module name (use command line)
+    (char*)cmd,               // Command line
+    NULL,                     // Process handle not inheritable
+    NULL,                     // Thread handle not inheritable
+    FALSE,                     // Set handle inheritance to FALSE
+    0,
+    NULL,                     // Use parent's environment block
+    NULL,                     // Use parent's starting directory 
+    &StartupInfo,             // Pointer to STARTUPINFO structure
+    &ProcessInformation       // Pointer to PROCESS_INFORMATION structure
+  );
+  CHECK_EQ(ret, TRUE, "CreateProcess failed");
+
+  // Wait until child process exits.
+  DWORD wait_ret = WaitForSingleObject(ProcessInformation.hProcess, INFINITE);
+  CHECK_EQ(wait_ret, WAIT_OBJECT_0, "Process ended strange");
```

Comment:
No, only on strage cases

Return code/value | Description |
----------------- | ----------- |
WAIT_ABANDONED 0x00000080L | The specified object is a mutex object that was not released by the thread that owned the mutex object before the owning thread terminated. Ownership of the mutex object is granted to the calling thread and the mutex state is set to nonsignaled. If the mutex was protecting persistent state information, you should check it for consistency.
WAIT_OBJECT_0 0x00000000L | The state of the specified object is signaled.
WAIT_TIMEOUT 0x00000102L | The time-out interval elapsed, and the object's state is nonsignaled.
WAIT_FAILED 0xFFFFFFFF | The function has failed. To get extended error information, call GetLastError.